ddl/column.go

```diff
@@ -308,6 +332,52 @@ func doModifyColumn(t *meta.Meta, job *model.Job, newCol *model.ColumnInfo, oldN
 	// }
 	// }
 
+	// Get sessionctx from context resource pool.
+	var ctx sessionctx.Context
+	ctx, err = w.sessPool.get()
+	if err != nil {
+		return ver, errors.Trace(err)
+	}
+	defer w.sessPool.put(ctx)
+
+	if !mysql.HasNotNullFlag(oldCol.Flag) && mysql.HasNotNullFlag(newCol.Flag) {
+		err = CheckForNullValue(ctx, dbInfo.Name, tblInfo.Name, oldCol.Name, newCol.Name)
+		if err != nil {
+			job.State = model.JobStateCancelled
```

Comment:
set `job.State = model.JobStateRollingback` for rollback to clear `PreventNullInsertFlag`